cub/cub/device/dispatch/tuning/tuning_three_way_partition.cuh

```diff
@@ -224,6 +224,112 @@ struct sm90_tuning<Input, OffsetT, input_size::_16, offset_size::_8>
   using delay_constructor                            = no_delay_constructor_t<1050>;
 };
 
+template <class InputT,
+          class OffsetT,
+          input_size InputSize   = classify_input_size<InputT>(),
+          offset_size OffsetSize = classify_offset_size<OffsetT>()>
+struct sm100_tuning;
+
+// template <class Input, class OffsetT>
+// struct sm100_tuning<Input, OffsetT, input_size::_1, offset_size::_4>
+// {
+//   // trp_0.ipt_12.tpb_256.ns_792.dcid_6.l2w_365 1.063960  0.978016  1.072833  1.301435
+//   static constexpr int items                         = 12;
+//   static constexpr int threads                       = 256;
+//   static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_DIRECT;
+//   using delay_constructor                            = exponential_backon_jitter_constructor_t<792, 365>;
+// };
+
+// template <class Input, class OffsetT>
+// struct sm100_tuning<Input, OffsetT, input_size::_2, offset_size::_4>
+// {
+//   // trp_1.ipt_14.tpb_288.ns_496.dcid_6.l2w_400 1.170449  1.123515  1.170428  1.252066
+//   static constexpr int items                         = 14;
+//   static constexpr int threads                       = 288;
+//   static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
+//   using delay_constructor                            = exponential_backon_jitter_constructor_t<496, 400>;
+// };
```

Comment:
Should we be explicit and name the same as SM90 as in other tunings?

Reply:
The more I think about it, the less I like constructs like:
```
// default back to SM90 tuning
template <....>
struct sm100_tuning<...> : sm90_tuning<...> {};
```
Tunings can work differently for each algorithm or architecture, so here `sm100_tuning` can have different template arguments than `sm90_tuning`, or different data members. Also, if `sm100_tuning` had a `::value` that could be interpreted differently than `sm90_tuning::value`, by the selection logic in the policy hub. Therefore, IMO, the best is to not provide a template specialization at all and let SFINAE not find an `sm100_tuning` and fall back.

But I could add a comment here.